### PR TITLE
Added domain comparing ability

### PR DESCRIPTION
**Pull request by H3JFC**
Hey @rverton, tell me what you think. This was pretty quick and dirty so I'm open to suggestions.

### CODE REVIEW

File: cmd/webanalyze/main.go [hunk 1]
```diff
@@ -32,6 +33,7 @@ func init() {
 	flag.StringVar(&host, "host", "", "single host to test")
 	flag.StringVar(&hosts, "hosts", "", "filename with hosts, one host per line.")
 	flag.IntVar(&crawlCount, "crawl", 0, "links to follow from the root page (default 0)")
+	flag.BoolVar(&searchSubdomain, "search", true, "searches all urls with same base domain (i.e. example.com and sub.exmaple.com)")
```
Comment (philipithomas) [inline cmd/webanalyze/main.go]: Misspelling: `sub.exmaple.com` -> `sub.example.com`

**Comment by philipithomas** [comment]
To clarify: this would be really helpful for us. We use `webanalyze` to scan company websites. A lot of sites host their application on a separate subdomain than their marketing site. (e.g., `www.example.com` might be a Wordpress site, and `app.example.com` might be a React site). We'd like to be able to understand the totality of technologies that a company is using.

**Comment by rverton** [comment]
Thanks for your contribution, I will have a look and test this as soon as possible.

**Comment by H3JFC** [comment]
Thanks Robin. It's much appreciated 👍

**Comment by rverton** [comment]
I made some minor changes to the variable name and fixed the tests. Thanks for your contribution!

**Pull request merged**